pinot-core/src/test/java/org/apache/pinot/core/transport/QueryRoutingTest.java

```diff
@@ -77,9 +78,16 @@ public void setUp() {
     _requestCount = 0;
   }
 
+  @AfterMethod
```

Comment:
Can we register metric `@BeforeClass` and deregister it `@AfterClass`? Also I don't know if we really need to deregister it

Reply:
I think this is the correct behavior. We need to deregister in order to add a new mock. We could reset the mock, but that would make it even more "mock dependent".

We could register the mock on a `@BeforeClass`, then clean it in a `@BeforeMethod`. That would be fine, but I would need to modify the test even more.